compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5927,10 +5837,149 @@ public void visit(BLangGroupExpr groupExpr) {
+                    symResolver.lookupClosureVarSymbol(encInvokableEnv, varSymbol.name, SymTag.VARIABLE);
+            if (resolvedSymbol != symTable.notFoundSymbol) {
+                varSymbol.closure = true;
+                ((BLangFunction) encInvokable).closureVarSymbols.add(new ClosureVarSymbol(varSymbol, pos));
+            }
+        }
+    }
+
+    private List<String> getNamesOfUserSpecifiedRecordFields(List<RecordLiteralNode.RecordField> userSpecifiedFields) {
+        List<String> fieldNames = new ArrayList<>();
+
+        for (RecordLiteralNode.RecordField field : userSpecifiedFields) {
+            if (field.isKeyValueField()) {
+                BLangExpression key = ((BLangRecordLiteral.BLangRecordKeyValueField) field).key.expr;
+                if (key.getKind() == NodeKind.LITERAL) {
+                    fieldNames.add(getLiteralKeyName(key));
+                } else if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                    fieldNames.add(getVariableKeyName(key));
+                }
+            } else if (field.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                fieldNames.add(getVariableFieldName(field));
+            } else {
+                addRequiredFieldsFromSpreadOperator(field, fieldNames);
+            }
+        }
+
+        return fieldNames;
+    }
+
+    private String getLiteralKeyName(BLangExpression key) {
+        return ((BLangLiteral) key).value.toString();
+    }
+
+    private String getVariableKeyName(BLangExpression key) {
+        return ((BLangSimpleVarRef) key).variableName.value;
+    }
+
+    private String getVariableFieldName(RecordLiteralNode.RecordField field) {
+        return ((BLangSimpleVarRef) field).variableName.value;
+    }
+
+    private void addRequiredFieldsFromSpreadOperator(RecordLiteralNode.RecordField field, List<String> fieldNames) {
+        BLangRecordLiteral.BLangRecordSpreadOperatorField spreadOpField =
+                (BLangRecordLiteral.BLangRecordSpreadOperatorField) field;
+        BType type = spreadOpField.expr.getBType();
+
+        if (type.tag == TypeTags.RECORD) {
+            for (BField bField : ((BRecordType) type).fields.values()) {
+                if (!Symbols.isFlagOn(bField.symbol.flags, Flags.REQUIRED)) {
+                    continue;
+                }
+                fieldNames.add(bField.name.value);
+            }
+        }
+    }
+
+    private void generateFieldsForUserUnspecifiedRecordFields(List<RecordLiteralNode.RecordField> fields,
+                                                              List<String> fieldNames,
+                                                              Map<String, BInvokableSymbol> defaultValues,
+                                                              Location pos, boolean isReadonly) {
+        for (Map.Entry<String, BInvokableSymbol> entry : defaultValues.entrySet()) {
+            String fieldName = entry.getKey();
+            if (!fieldNames.contains(fieldName)) {
```

Comment:
Check inverse and continue early.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/37184/commits/f846176e6ff38c3e1366f21da4cfa05ea8a4fbdc